markdown/exploring-solace-native-apis/exploring-solace-native-apis.md

```diff
@@ -0,0 +1,601 @@
+
+The Java API has several message types that you can use to make manipulating message payloads easier.  You'll notice from before that we create a **TextMessage** - you could just as easily create a BytesMessage, for instance.
+
+#### Adding the contents to the message
+
+In Java, we simply call the appropriate function for the message object we've created.  In TopicPublisher, we've create a TextMessage, so:
+
+	msg.setText(text);
+
+In C, we manipulate the **Binary Attachment**:
+
+	solClient_msg_setBinaryAttachment ( msg_p, text_p, ( solClient_uint32_t ) strlen ( (char *)text_p ) );
+
+###5 Send!
+
+In Java, we tell the producer where to send the message:
+
+	prod.send(msg,topic);
+
+In C, we've already annotated the message with the destination, so we just call send:
+
+	solClient_session_sendMsg ( session_p, msg_p );
+
+So, let's do that.  Firstly, let's set up our SDKPerf subscriber so we can see the message being received:
+
+	sdkperf -cip <broker ip> -cu <username>@<message-vpn> -stl tutorial/topic -md
+
+The "-md" stands for "message dump."
+
+Positive
+: I'd encourage you to always attach an sdkperf subsriber with message dump turned on whenever you're debugging an application.  It's tempting to write some code to probe the event broker and find out what's going on, but SDKPerf is the debug tool of choice for our support team and with good reason.  You'll see exactly what is being sent and received.
+
+Then we'll just fire up TopicPubisher and see what comes out:
+
+	^^^^^^^^^^^^^^^^^^ Start Message ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+	Destination:                            Topic 'tutorial/topic'
+	Priority:                               4
+	Class Of Service:                       USER_COS_1
+	DeliveryMode:                           DIRECT
+	Message Id:                             6
+	Binary Attachment:                      len=15
+	1c 0f 48 65 6c 6c 6f 20    77 6f 72 6c 64 21 00       ..Hello.world!.
+	
+	^^^^^^^^^^^^^^^^^^ End Message ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This is what TopicPublisher sent.  Since it's what we were hoping to send, we can now attach our own subscribing application.  If only we had one... hold on - TopicSubscriber!
+
+###6 Publish/Subscribe
+
+**LEAVE SDKPERF RUNNING** and run TopicSubscriber.  The re-run topic publisher.  What you should see is that same message appearing in both applications.  By the magic of Publish/Subscribe, we've sent a single message to two different applications *without either of the original sender or receiver being aware of it*.
+
+![](./img/part-c-2.png)
+
+You could even download and run different TopicSubscribers from different APIs, or different SolaceSamples.  That shows how you can send a message in one language and receive it in different languages.
+
+## Wrap up, and Things We Didn't Cover
+Duration 0:03:00
```

Comment:
you forgot a `:` 
`Duration: 0:03:00`
![Screen Shot 2020-06-22 at 5 08 01 PM](https://user-images.githubusercontent.com/6826053/85335724-75facf00-b4ab-11ea-8b18-16b3bbdde24e.png)